### PR TITLE
lib/config, lib/fs: Make junction behaviour configurable (ref #6606)

### PR DESCRIPTION
**Pull request by imsodin**
See https://forum.syncthing.net/t/option-to-follow-directory-junctions-symbolic-links/14750/126 and following.

### CODE REVIEW

**Comment by AudriusButkevicius** [comment]
I think we do a config migration of this, setting to true on all existing folders, and defaulting to false.

**Comment by imsodin** [comment]
Passing fs options as url-like query parameters might not be the best idea: A filesystem path does not necessarily conform to what `url.Parse` expects. An alternative is adding the usual `...opts Option` thing to `NewFilesystem` and then have `WithJunctionsAsDirs() Option`, though I believe Audrius has a bit of a disliking to that pattern (?).

**Comment by AudriusButkevicius** [comment]
I don't mind options, I mind functions thrown around like they are variables, yet I think query string stuff is the right way to go here.

**Comment by imsodin** [comment]
Well these options are functions disguised as options ;)  

Anyway my problem is that almost anything is allowed in paths, i.e. you need to do parsing yourself (`url.Parse` fails). At which point I'd rather not do the detour of encoding options into a pseudo-uri and then decoding the same options from that pseudo-uri, but pass them directly.

**Comment by AudriusButkevicius** [comment]
so my original idea was to turn C:\foo\bar into file://c/foo/bar, but it never happened. If it did happen, this would be much easier.

This also would have allowed getting rid fo the type, as the schema dictates the type.

**Comment by imsodin** [comment]
That makes sense, but being an unrelated change and quite possible to turn up weird cornercases, I used options, It's still easy enough to switch to "all uri initialisation" from here.

**Review by AudriusButkevicius — DISMISSED** [review]

File: lib/fs/basicfs.go [hunk 1]
```diff
@@ -23,13 +23,20 @@ var (
 	ErrNotRelative     = errors.New("not a relative path")
 )
 
+func WithJunctionsAsDirs() Option {
+	return func(fs Filesystem) {
+		fs.(*BasicFilesystem).junctionsAsDirs = true
```
Comment (calmh) [inline lib/fs/basicfs.go:28]: Panic if I ask for a fakefs with junctionsAsDirs?

**Review by AudriusButkevicius — APPROVED** [review]

**Review by calmh — APPROVED** [review]

**Pull request merged**